inference-engine/tests/functional/plugin/cpu/single_layer_tests/one_hot.cpp

```diff
@@ -3,165 +3,330 @@
-        auto inputParams = ngraph::builder::makeParams(ngPrc, { inputShape });
+        init_input_shapes({inputShape});
+        if (inputType.second) {
+            for (auto &target : targetStaticShapes)
+                target.push_back({});
+        }
 
-        auto oneHot = std::make_shared<ngraph::opset5::OneHot>(inputParams.front(), depthConst, onConst, offConst, axis);
-        function = makeNgraphFunction(ngPrc, inputParams, oneHot, "OneHot");
+        function = createFunction(inputType.first == ngraph::helpers::InputLayerType::CONSTANT);
+    }
+    void init_ref_function(std::shared_ptr<ov::Function> &funcRef, const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        if (function->get_parameters().size() == 2) {
+            generateDepth();
+            funcRef = createFunction(true);
+        }
+        ngraph::helpers::resize_function(funcRef, targetInputStaticShapes);
+    }
+    void validate() override {
+        if (function->get_parameters().size() == 2) {
+            auto pos = std::find_if(inputs.begin(), inputs.end(),
+                                    [](const std::pair<std::shared_ptr<ov::Node>, ov::runtime::Tensor> &params) {
+                                        return params.first->get_friendly_name() == "ParamDepth";
+                                    });
+            IE_ASSERT(pos != inputs.end());
+            inputs.erase(pos);
+        }
+        SubgraphBaseTest::validate();
+    }
+    std::shared_ptr<ngraph::Function> createFunction(bool depthConst) {
+        auto params = ngraph::builder::makeDynamicParams(ngraph::element::i32, {inputDynamicShapes.front()});
+        params.front()->set_friendly_name("ParamsIndices");
+        std::shared_ptr<ov::Node> depth;
+        if (depthConst) {
+            depth = ngraph::op::Constant::create(ngraph::element::i32, ngraph::Shape{ }, {Depth});
+        } else {
+            auto depthParam = std::make_shared<ngraph::op::Parameter>(ngraph::element::i32, ngraph::Shape{ });
+            depthParam->set_friendly_name("ParamDepth");
+            params.push_back(depthParam);
+            depth = depthParam;
+        }
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::opset3::Parameter>(params));
+        auto on_value_const = std::make_shared<ngraph::op::Constant>(outType, ngraph::Shape{ }, OnValue);
+        auto off_value_const = std::make_shared<ngraph::op::Constant>(outType, ngraph::Shape{ }, OffValue);
+        auto oneHot = std::make_shared<ngraph::opset5::OneHot>(paramOuts[0], depth, on_value_const, off_value_const, Axis);
+        return makeNgraphFunction(ngraph::element::i32, params, oneHot, "OneHot");
     }
+    void generateDepth() {
+        ov::runtime::Tensor tensor = ov::test::utils::create_and_fill_tensor(ov::element::Type_t::i32, {}, 10, 1, 1, time(0));
+        auto *dataPtr = tensor.data<int32_t>();
```

Comment:
I mean can we use `testing::internal::Random` to generate values

https://github.com/openvinotoolkit/openvino/pull/8488#discussion_r750039048